python/pywatchman/__init__.py

```diff
@@ -690,10 +690,17 @@ def _resolvesockname(self):
 
         cmd = ['watchman', '--output-encoding=bser', 'get-sockname']
         try:
+            # if invoked via an application with a graphical user interface,
+            # this call will cause a brief command window pop-up.
+            # Using the flag STARTF_USESHOWWINDOW to avoid this behavior.
+            startupinfo = subprocess.STARTUPINFO()
+            startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
```

Comment:
This won't be available on Unix platforms, right? It'd be worth factoring this out into a separate function and doing different things on Windows vs Unix.